src/test/java/org/broadinstitute/hellbender/tools/walkers/bqsr/ApplyBQSRIntegrationTest.java

```diff
@@ -82,26 +86,31 @@ public String getTestedClassName() {
 
     @Test(dataProvider = "ApplyBQSRTest")
     public void testApplyBQSR(ABQSRTest params) throws IOException {
-        File outFile = GATKBaseTest.createTempFile("applyBQSRTest", params.outputExtension);
-        final ArrayList<String> args = new ArrayList<>();
-        File refFile = null;
-
-        args.add("-I");
-        args.add(new File(params.bam).getAbsolutePath());
-        args.add("--" + StandardArgumentDefinitions.BQSR_TABLE_LONG_NAME);
-        args.add(new File(resourceDir + "HiSeq.20mb.1RG.table.gz").getAbsolutePath());
-        args.add("-O"); args.add(outFile.getAbsolutePath());
-        if (params.reference != null) {
-            refFile = new File(params.reference);
-            args.add("-R"); args.add(refFile.getAbsolutePath());
+        try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
```

Comment:
Can you keep the existing test code as-is, and make the Jimfs-based tests a separate test case? It would also be good to have one test case that writes to a live GCS bucket (marked as `groups = {bucket}`)

Reply:
Re-instated the old (File-based) test. Remains to add a GCS test.

Reply:
GCS test added. For now it runs all the cases, but since the other tests already establish that ApplyBQSR does the right thing and we just want to make sure it can write to GCS, it'd make sense to only run a subset of the cases on the cloud. If you let me know which ones you want, I can cull the rest.